launcher/helper.cpp

```diff
@@ -89,8 +89,36 @@ QString getRealPath(QString path)
 void performNativeCopy(QString src, QString dst)
 {
 #ifdef VCMI_ANDROID
-	auto srcStr = QAndroidJniObject::fromString(src);
-	auto dstStr = QAndroidJniObject::fromString(dst);
+	auto percentEncodeNonAscii = [](const QString &input) {
+		QByteArray utf8 = input.toUtf8();
+		QByteArray encoded;
+
+		for (char c : utf8)
```

Comment:
seems easier to do this instead of casting later:
```suggestion
		for (unsigned char c : utf8)
```